include/hw/riscv/boot.h

```diff
@@ -21,10 +21,13 @@
 #ifndef HW_RISCV_BOOT_H
 #define HW_RISCV_BOOT_H
 
-hwaddr riscv_load_firmware(const char *filename);
-hwaddr riscv_load_kernel(const char *filename, void *fdt);
+hwaddr riscv_load_firmware(const char *filename, hwaddr ram_start,
+                           uint64_t ram_size);
+hwaddr riscv_load_kernel(const char *filename, void *fdt, hwaddr ram_start,
+                         uint64_t ram_size);
 void riscv_load_initrd(const char *filename, uint64_t mem_size,
                        hwaddr firmware_entry, void *fdt);
-hwaddr riscv_load_firmware_kernel_initrd(MachineState *machine, void *fdt);
+hwaddr riscv_load_firmware_kernel_initrd(MachineState *machine, void *fdt,
+                                         hwaddr ram_start);
```

Comment:
The bulk of the patch seems to be plumbing through ram_size. The actual change is pretty small after this unnecessary plumbing is removed.

Reply:
Thanks for your review!

That's true, but I will still need to add ram_start here.